projects/step-frontend/src/lib/plugins-initializer/plugins-initializer.ts

```diff
@@ -24,34 +28,32 @@ const fetchDefinitions = async (): Promise<PluginDefinition[]> => {
       const angularModules = (plugin as LegacyPluginDefinition)?.angularModules || [];
       const toOverride = angularModules.find((m) => OVERRIDE_PLUGINS.has(m));
       if (toOverride) {
+        const name = toOverride;
         const entryPoint = OVERRIDE_PLUGINS.get(toOverride)!;
-        console.log(`Plugin "${toOverride}" configuration was overridden to the new format`);
-        return { entryPoint };
+        console.log(`Plugin "${name}" configuration was overridden to the new format`);
+        return { entryPoint, name };
       }
 
       return plugin;
     });
 
     if (ADDITIONAL_PLUGINS.length > 0) {
-      const additionalPlugins: MicrofrontendPluginDefinition[] = ADDITIONAL_PLUGINS.map((entryPoint) => ({
-        entryPoint,
-      }));
-      result = [...result, ...additionalPlugins];
+      result = [...result, ...ADDITIONAL_PLUGINS];
     }
   } catch (e) {
     console.log('Fetch plugin definitions failed', e);
   }
   return result;
 };
 
-const loadPlugins = (compiler: Compiler, injector: Injector) => {
+const loadPlugins = (compiler: Compiler, injector: Injector, registry: PluginInfoRegistryService) => {
   return async () => {
     const pluginDefinitions = await fetchDefinitions();
     if (pluginDefinitions.length === 0) {
       return;
     }
 
-    const { legacy, microfrontend } = pluginDefinitions.reduce(
+    let { legacy, microfrontend } = pluginDefinitions.reduce(
```

Comment:
I know microfrontend is not a new variable/class but it is a bit dificult to understand for me. What exactly is the name standing for?

Reply:
We have to types LegacyPluginDefinition and MicrofrontendPluginDefinition
We receive an array pluginDefenitions, that may contain elements of both of these types.
Then the reduce is called, and aggregates items of each type to corresponding array - `legacy` for LegacyPluginDefinition items and `microfrontend ` for MicrofrontendPluginDefinition items